src/cli/rustup_mode.rs

```diff
@@ -807,7 +808,7 @@ fn show(cfg: &Cfg) -> Result<()> {
         }
     }
 
-    fn print_header(t: &mut term2::Terminal<std::io::Stdout>, s: &str) -> Result<()> {
+    fn print_header(t: &mut term::StdoutTerminal, s: &str) -> Result<()> {
```

Comment:
Should this not be `term2::StdoutTerminal` to reduce leakage of `term` api?

Reply:
I don't think so; thats a concrete type with the isatty and error silencing built in (which BTW is really showing where we abuse the API and call e.g. reset() without checking supports_reset() .... but thats a different story); anyhow - point is that print_header doesn't care about that distinction, and it was giving it the concrete type in the past that was encouraging the markdown comingling in the first place. So I want to make it clearer:

this depends on 'a terminal'

this implements our terminal specialisations.

And I think depending on term2:StdoutTerminal gets that wrong because of the current type alias.

I did have a patch where I ditched that entirely and term2:StdoutTerminal was an export of term:StdoutTerminal, and stdio() was returning a Box, but I decided that that didn't add much.

Reply:
Aha, right, could we have an `impl` in there then (i.e. `&mut impl term::StdoutTerminal`) just to make it clear that this is a trait not a struct we're getting a borrow to?

Reply:
mm, uh,so actually it is a struct we're passing in, not a trait object, so I dunno. Can we just declare it bankrupt as we're about to rip out term anyway?

Reply:
Oh, argh, yeah OK